src/platform/datapath_win.c

```diff
@@ -0,0 +1,628 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    QUIC Datapath Implementation (User Mode)
+
+--*/
+
+#include "platform_internal.h"
+
+#ifdef QUIC_CLOG
+#include "datapath_winuser.c.clog.h"
+#endif
+
+#pragma warning(disable:4116) // unnamed type definition in parentheses
+
+#pragma warning(disable:4100) // unreferenced
+#pragma warning(disable:6101) // uninitialized
+#pragma warning(disable:6386) // buffer overrun
+
+#define IS_LOOPBACK(Address) ((Address.si_family == QUIC_ADDRESS_FAMILY_INET &&                \
+                               Address.Ipv4.sin_addr.S_un.S_addr == htonl(INADDR_LOOPBACK)) || \
+                              (Address.si_family == QUIC_ADDRESS_FAMILY_INET6 &&               \
+                               IN6_IS_ADDR_LOOPBACK(&Address.Ipv6.sin6_addr)))
+
+#define DatapathType(SendData) ((CXPLAT_SEND_DATA_COMMON*)(SendData))->DatapathType
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+CxPlatDataPathInitialize(
+    _In_ uint32_t ClientRecvContextLength,
+    _In_opt_ const CXPLAT_UDP_DATAPATH_CALLBACKS* UdpCallbacks,
+    _In_opt_ const CXPLAT_TCP_DATAPATH_CALLBACKS* TcpCallbacks,
+    _In_opt_ QUIC_EXECUTION_CONFIG* Config,
+    _Out_ CXPLAT_DATAPATH** NewDataPath
+    )
+{
+    QUIC_STATUS Status = QUIC_STATUS_SUCCESS;
+    if (NewDataPath == NULL) {
+        Status = QUIC_STATUS_INVALID_PARAMETER;
+        goto Error;
+    }
+
+    Status = DataPathInitialize(
+        ClientRecvContextLength,
+        UdpCallbacks,
+        TcpCallbacks,
+        Config,
+        NewDataPath);
```

Comment:
We use the following style:
```suggestion
    Status =
        DataPathInitialize(
            ClientRecvContextLength,
            UdpCallbacks,
            TcpCallbacks,
            Config,
            NewDataPath);
```